test/integration/tcp_proxy_integration_test.h

```diff
@@ -12,17 +12,13 @@ namespace {
 class TcpProxyIntegrationTest : public BaseIntegrationTest,
                                 public testing::TestWithParam<Network::Address::IpVersion> {
 public:
-  TcpProxyIntegrationTest() : BaseIntegrationTest(GetParam()) {}
+  TcpProxyIntegrationTest()
+      : BaseIntegrationTest(GetParam(), Http::CodecClient::Type::HTTP1,
```

Comment:
It seems like BaseIntegrationTest shouldn't know anything about HTTP at all, though I'm not sure how practical that is. Not a big deal if hard.

Reply:
Sure, that can go back to the Http test.

We still have HTTP in the base integration test since the fake upstreams are created with HTTP1 but that's preexisting in the TCP proxy tests so I think it's out of scope for this refactor.

Reply:
It's not a big deal either way, just thought I would point it out.